src/item.cpp

```diff
@@ -4033,14 +4020,12 @@ int item::getlight_emit() const {
     if ( has_flag("CHARGEDIM") && is_tool() && !has_flag("USE_UPS")) {
         it_tool * tool = dynamic_cast<it_tool *>(type);
         int maxcharge = tool->max_charges;
-        if ( maxcharge > 0 ) {
-            lumint = ( type->light_emission * chargedrop * charges ) / maxcharge;
+        // Falloff starts at 1/5 total charge and scales linearly from there to 0.
+        if( maxcharge > 0 && charges > 0 && maxcharge / charges > 5 ) {
+            lumint *= charges * 5 / maxcharge;
```

Comment:
I thinks this is responsible for the flashlight bug Coolthulhu mentioned. When charges is 17 (maxcharges is 100), the if-condition is not met, but as soon as it drops to 16 charges, it is. And then `charge * 5 / maxcharges` evaluates to 0 (a `long`) all the time.